libs/api/books/src/lib/reading-list.service.ts

```diff
@@ -28,4 +28,22 @@ export class ReadingListService {
       return list.filter(x => x.bookId !== id);
     });
   }
+
+  async markBookFinished(item: ReadingListItem): Promise<void> {
+    const index = this.storage.read().findIndex(book => book.bookId === item.bookId);
+    this.storage.update(list => {
```

Comment:
You could do this with a single operation -> `this.storage.update(list => list.map(...`